src/vse_sync_pp/analyzers/ppsdpll.py

```diff
@@ -0,0 +1,209 @@
+### SPDX-License-Identifier: GPL-2.0-or-later
+
+import pandas as pd
+import numpy as np
+import allantools
+
+from collections import namedtuple
+from decimal import Decimal
+
+class Analyzer():
+    def __init__(self):
+        self._col1_name = 'timestamp'
+        self._col2_name = 'eecstate'
+        self._col3_name = 'phasestate'
+        self._col4_name = 'phaseoffset'
+        self._data  = pd.DataFrame(columns=['timestamp','eecstate','phasestate','phaseoffset']) 
+    def append(self, data):
+        """Append x and y data points extracted from `data`"""
+        col1_val = getattr(data, self._col1_name)
+        col2_val = getattr(data, self._col2_name)
+        col3_val = getattr(data, self._col3_name)
+        col4_val = getattr(data, self._col4_name)
+        self._data.loc[len(self._data)]=[col1_val,col2_val,col3_val,float(col4_val)]
+    def retrieve(self):
+        """Return the data in Data Frame format"""
+        return self._data
+    def to_visualize_stage(self, col1, col2, col3, col4):
+        elems = (self._col1_name, self._col2_name, self._col3_name, self._col4_name)
+        data = namedtuple('Analyzed', elems)
```

Comment:
I would define this a head of time as your going to repeatedly call this function.